include/openthread/border_agent.h

```diff
@@ -282,6 +282,16 @@ void otBorderAgentSetEphemeralKeyCallback(otInstance                       *aIns
                                           otBorderAgentEphemeralKeyCallback aCallback,
                                           void                             *aContext);
 
+/**
+ * Disconnects the Border Agent from any active secure sessions. If Border Agent is connected
+ * to a commissioner candidate with ephemeral key, calling this api will cause the ephemeral key to
+ * be cleared.
+ *
```

Comment:
It may be helpful to add a comment here that the BA state may not change immediately upon calling the `otBorderAgentDisconnect()`. Callers of this API might assume otherwise.

The BA needs to wait for the `HandleConnected()` callback from the DTLS secure transport, and there's a guard delay of `kGuardTimeNewConnectionMilli = 2000` in the code:

```c++
void SecureTransport::Disconnect(ConnectEvent aEvent)
{
    VerifyOrExit(IsStateConnectingOrConnected());

    mbedtls_ssl_close_notify(&mSsl);
    SetState(kStateCloseNotify);
    mConnectEvent = aEvent;
    mTimer.Start(kGuardTimeNewConnectionMilli);

    mMessageInfo.Clear();

    FreeMbedtls();

exit:
    return;
}
```

Reply:
Thanks, updated the doc, PTAL

Reply:
Shouldn't this API forces the disconnection without waiting for the timer?

Reply:
I expect this API should be terminated all active sessions immediately.

Reply:
The disconnection happens immediately, but the notification of the disconnection happens after a guard delay of 2s. Not sure about the intention of the guard delay, maybe we can make the timer fire immediately for locally closed connections, but I'm not sure if this would break anything. Thoughts?

Reply:
Yes, just checked the session is closed immediately. That should be sufficient.

Reply:
Thanks @sunytt for adding this. LGTM.

@bukepo @wgtdkp, to clarify, this is existing behavior in the code. I recently read/reviewed this part of the code and therefore recalled this behavior.

I asked this to be mentioned to avoid surprises and confusion about the behavior of this API (which your comment indicates such a expectation may indeed arise).

I don't know why this delay was added in the first place to `secure_transport.cpp` (formerly `dtls.cpp`). It seems logical to investigate this to understand its purpose and potentially modify it if it is unnecessary. Two second delay does not seems reasonable to me. However, I'd say this is a completely separate topic than this PR. Do either of you happen to know the reason (and/or may want to take the lead on this)?